src/main/java/ogc/rs/apiserver/ApiServerVerticle.java

```diff
@@ -1919,62 +1919,146 @@ public void getAssets(RoutingContext routingContext) {
+        String id = collection.getString("id");
+
+        if (!idSet.add(id)) {
+          OgcException ogcException =
+              new OgcException(400, "Bad Request", "Duplicate Ids present in request body.");
+          routingContext.put("response", ogcException.getJson().toString());
+          routingContext.put("statusCode", ogcException.getStatusCode());
+          routingContext.next();
+          return;
+
+          }
+
+        collection
+            .put("accessPolicy", routingContext.data().get("accessPolicy"))
+            .put("ownerUserId", routingContext.data().get("ownerUserId"))
+            .put("role", routingContext.data().get("role"));
+
+        collectionNew.add(collection);
+      }
+
+      dbService
+          .postStacCollections(collectionNew)
+          .onSuccess(
+              dbRequest -> {
+                JsonArray responseBody = new JsonArray();
+                collections.forEach(
+                    obj -> responseBody.add(processStacCollection((JsonObject) obj)));
+
+                routingContext.put("response", responseBody.toString()).put("statusCode", 201);
+                routingContext.next();
+              })
+          .onFailure(
+              failed -> {
+                String errorMessage = failed.getMessage();
+                JsonObject errorJson;
+
+                try {
+                  errorJson = new JsonObject(errorMessage);
+                  JsonArray failedId = errorJson.getJsonArray("failedIds", new JsonArray());
+                  LOGGER.error(
+                      "STAC Collection Onboarding Failed for id: "
+                          + failedId
+                          + " due to "
+                          + errorJson.getString("description"));
+
+                  int statusCode = 400;
+                  if (errorJson
+                      .getString("description")
+                      .contains("duplicate key value violates unique constraint")) {
+                    statusCode = 409;
+                  }
+                  OgcException ogcException =
+                      new OgcException(
+                          statusCode,
+                          "Conflict",
+                          "One or more Collection-id(s) already exists!");
+                  routingContext.put("response", ogcException.getJson().toString());
+                  routingContext.put("statusCode", ogcException.getStatusCode());
+                  routingContext.next();
+                } catch (Exception e) {
+                  LOGGER.debug("Failed to parse error JSON. Using plain error message.");
+                  OgcException ogcException =
+                      new OgcException(500, "Internal Server Error ", errorMessage);
+                  routingContext.put("response", ogcException.getJson().toString());
+                  routingContext.put("statusCode", ogcException.getStatusCode());
+                  routingContext.next();
+                }
+              });
```

Comment:
For the `onFailure` cases for both single and multiple collections, can you see if it's possible to return the `OgcException` directly from the `dbService` and not do any string matching for handling different errors.